File: 2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.py
class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        res = 0
        for i in range(32):
            count = 0
            for n in candidates:
                count += 1 if (1 << i) & n else 0
            res = max(res, count)
        return res
        # counts = [0] * 32

        # for n in candidates:
        #     i = 0
        #     while n:
        #         counts[i] += n & 1
        #         n = n >> 1
        #         i += 1
        # return max(counts)
            
        

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int result = 0;
        for (int i = 0; i < 32; i++){
            int count = 0;
            for (int n: candidates){
                count += (1 << i & n) > 0 ? 1: 0;
            }
            result = max(result, count);
        }
        return result;

        // vector<int> count(32, 0);
        // for (int n: candidates){
        //     int i = 0;
        //     while (n > 0){
        //         count[i] += n & 1;
        //         n = n >> 1;
        //         i += 1;
        //     }
        // }
        // int result = 0;
        // for (int n: count)
        //     if (result < n) result = n;
        // return result;
    }
};